src/HealthChecks.Nats/NatsHealthCheck.cs

```diff
@@ -28,9 +28,25 @@ public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, Canc
     {
         try
         {
-            _connection ??= CreateConnection(_options);
-            Guard.ThrowIfNull(_connection);
-            var healthCheckResult = GetHealthCheckResultFromState(_connection);
+            // Create new connection if there is no existing one
+            IConnection? connection = _connection;
+            if (connection == null)
+            {
+                connection = CreateConnection(_options);
+                var exchanged = Interlocked.CompareExchange(ref _connection, connection, null);
+                if (exchanged != null) // was set by other thread
+                {
+                    connection.Dispose();
+                    connection = exchanged;
+                }
+            }
+
+            // reset connection in case of stuck so the next HC call will establish it again
+            // https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/issues/1544
+            if (connection.State == ConnState.DISCONNECTED || connection.State == ConnState.CLOSED)
+                _connection = null;
```

Comment:
When this branch is executed (setting `connection` to null) then the code afterwards may throw `System.NullReferenceException` because it's accessing the `State` property in the `GetHealthCheckResultFromState` method.

Reply:
_connection is not connection